src/power_grid_model_io/converters/pandapower_converter.py

```diff
@@ -0,0 +1,460 @@
+        pp_trafo3w["index"] = pp_trafo3w.index
+
+        # Select the appropriate switches and columns
+        pp_switches = self.pp_data["switch"]
+        pp_switches = pp_switches[self.pp_data["switch"]["et"] == "t3"]
+        pp_switches = pp_switches[["element", "bus", "closed"]]
+
+        # Join the switches with the 3wtrafos three times, for the hv_bus, for the mv_bus and once for the lv_bus
+        pp_1_switches = (
+            pp_trafo3w[["index", "hv_bus"]]
+            .merge(
+                pp_switches,
+                how="left",
+                left_on=["index", "hv_bus"],
+                right_on=["element", "bus"],
+            )
+            .fillna(True)
+            .set_index(pp_trafo3w.index)
+        )
+        pp_2_switches = (
+            pp_trafo3w[["index", "mv_bus"]]
+            .merge(
+                pp_switches,
+                how="left",
+                left_on=["index", "mv_bus"],
+                right_on=["element", "bus"],
+            )
+            .fillna(True)
+            .set_index(pp_trafo3w.index)
+        )
+        pp_3_switches = (
+            pp_trafo3w[["index", "lv_bus"]]
+            .merge(
+                pp_switches,
+                how="left",
+                left_on=["index", "lv_bus"],
+                right_on=["element", "bus"],
+            )
+            .fillna(True)
+            .set_index(pp_trafo3w.index)
+        )
+
+        pgm_3wtransformers = initialize_array(
+            data_type="input", component_type="three_winding_transformer", shape=len(pp_trafo3w)
+        )
+        pgm_3wtransformers["id"] = self._generate_ids("trafo3w", pp_trafo3w.index)
+        pgm_3wtransformers["node_1"] = self._get_ids("bus", pp_trafo3w["hv_bus"])
+        pgm_3wtransformers["node_2"] = self._get_ids("bus", pp_trafo3w["mv_bus"])
+        pgm_3wtransformers["node_3"] = self._get_ids("bus", pp_trafo3w["lv_bus"])
+        pgm_3wtransformers["status_1"] = pp_trafo3w["in_service"] & pp_1_switches["closed"]
+        pgm_3wtransformers["status_2"] = pp_trafo3w["in_service"] & pp_2_switches["closed"]
+        pgm_3wtransformers["status_3"] = pp_trafo3w["in_service"] & pp_3_switches["closed"]
+        pgm_3wtransformers["u1"] = pp_trafo3w["vn_hv_kv"] * 1e3
+        pgm_3wtransformers["u2"] = pp_trafo3w["vn_mv_kv"] * 1e3
+        pgm_3wtransformers["u3"] = pp_trafo3w["vn_lv_kv"] * 1e3
+        pgm_3wtransformers["sn_1"] = pp_trafo3w["sn_hv_mva"] * 1e6
+        pgm_3wtransformers["sn_2"] = pp_trafo3w["sn_mv_mva"] * 1e6
+        pgm_3wtransformers["sn_3"] = pp_trafo3w["sn_lv_mva"] * 1e6
+        pgm_3wtransformers["uk_12"] = pp_trafo3w["vk_hv_percent"] * 1e-2
+        pgm_3wtransformers["uk_13"] = pp_trafo3w["vk_lv_percent"] * 1e-2
+        pgm_3wtransformers["uk_23"] = pp_trafo3w["vk_mv_percent"] * 1e-2
+
+        pgm_3wtransformers["pk_12"] = (pp_trafo3w["vkr_hv_percent"] * 1e-2) * (
```

Comment:
You could combine the multiplications. Right now you have

```python
pk = (vkr * 1e-2) * (min(sn_hv, sn_mv) * 1e6)
```
Which is 3 x N multiplications, while this is the same as
```python
pk = vkr * min(sn_hv, sn_mv) * (1e-2 * 1e6)
```
Which is 2 x N + 1 multiplications